main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala

```diff
@@ -292,358 +292,353 @@ object Monomorph {
-      case Pattern.Wild(tpe, loc) => (Pattern.Wild(subst(tpe), loc), Map.empty)
-      case Pattern.Var(sym, tpe, loc) =>
-        // Generate a fresh variable symbol for the pattern-bound variable.
-        val freshSym = Symbol.freshVarSym(sym)
-        (Pattern.Var(freshSym, subst(tpe), loc), Map(sym -> freshSym))
-      case Pattern.Cst(cst, tpe, loc) => (Pattern.Cst(cst, tpe, loc), Map.empty)
-      case Pattern.Tag(sym, pat, tpe, loc) =>
-        val (p, env1) = visitPat(pat, subst)
-        (Pattern.Tag(sym, p, subst(tpe), loc), env1)
-      case Pattern.Tuple(elms, tpe, loc) =>
-        val (ps, envs) = elms.map(p => visitPat(p, subst)).unzip
-        (Pattern.Tuple(ps, subst(tpe), loc), envs.reduce(_ ++ _))
-    }
+    case Expression.Lazy(exp, tpe, loc) =>
+      val e = visitExp(exp, env0, subst)
+      Expression.Lazy(e, subst(tpe), loc)
 
-    def visitJvmMethod(method: JvmMethod, env0: Map[Symbol.VarSym, Symbol.VarSym], subst: StrictSubstitution) = method match {
-      case JvmMethod(ident, fparams0, exp0, tpe, pur, loc) =>
-        val (fparams, env1) = specializeFormalParams(fparams0, subst0)
-        val exp = visitExp(exp0, env0 ++ env1, subst)
-        JvmMethod(ident, fparams, exp, subst(tpe), pur, loc)
-    }
+    case Expression.Force(exp, tpe, pur, loc) =>
+      val e = visitExp(exp, env0, subst)
+      Expression.Force(e, subst(tpe), pur, loc)
+  }
+
+  /**
+    * Specializes the given pattern `p0` w.r.t. the current substitution.
+    *
+    * Returns the new pattern and a mapping from variable symbols to fresh variable symbols.
+    */
+  private def visitPat(p0: Pattern, subst: StrictSubstitution)(implicit flix: Flix): (Pattern, Map[Symbol.VarSym, Symbol.VarSym]) = p0 match {
+    case Pattern.Wild(tpe, loc) => (Pattern.Wild(subst(tpe), loc), Map.empty)
+    case Pattern.Var(sym, tpe, loc) =>
+      // Generate a fresh variable symbol for the pattern-bound variable.
+      val freshSym = Symbol.freshVarSym(sym)
+      (Pattern.Var(freshSym, subst(tpe), loc), Map(sym -> freshSym))
+    case Pattern.Cst(cst, tpe, loc) => (Pattern.Cst(cst, tpe, loc), Map.empty)
+    case Pattern.Tag(sym, pat, tpe, loc) =>
+      val (p, env1) = visitPat(pat, subst)
+      (Pattern.Tag(sym, p, subst(tpe), loc), env1)
+    case Pattern.Tuple(elms, tpe, loc) =>
+      val (ps, envs) = elms.map(p => visitPat(p, subst)).unzip
+      (Pattern.Tuple(ps, subst(tpe), loc), envs.reduce(_ ++ _))
+  }
 
-    visitExp(exp0, env0, subst0)
+  /**
+    * Specializes the given method `method` w.r.t. the current substitution.
+    *
+    * Returns the new method.
+    */
+  private def visitJvmMethod(method: JvmMethod, env0: Map[Symbol.VarSym, Symbol.VarSym], subst: StrictSubstitution)(implicit def2def: Def2Def, defQueue: DefQueue, root: Root, flix: Flix): JvmMethod = method match {
+    case JvmMethod(ident, fparams0, exp0, tpe, pur, loc) =>
+      val (fparams, env1) = specializeFormalParams(fparams0, subst)
```

Comment:
This `subst` here was `subst0` before which I believe is a typo (`subst0` would be the argument to specialize instead of the concrete substitution for this function call)

Reply:
Ok. Seems reasonable.